lib/rspec/support/reentrant_mutex.rb

```diff
@@ -25,10 +25,18 @@ def synchronize
         exit
       end
 
+      def locked?
+        @owner != nil
+      end
+
+      def owned?
+        @owner == Thread.current
+      end
```

Comment:
`ReentrantMutex` is a wrapper on top of either the native Ruby `Mutex` class of the RSpec version ([initialization](https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/reentrant_mutex.rb#L18) and [definition](https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/reentrant_mutex.rb#L44-L51)) so this should be the only place changes are required.